backend.py

```diff
@@ -42,6 +45,10 @@ def move(self, direction, distance, state):
                 (new_x, new_y) = direction.coor_delta
                 x = x + new_x
                 y = y + new_y
+                # When the robot leaves the map, robot dies
+                if (x, y) not in state.board:
+                    HoleTile.kill_robot(HoleTile, self, state)
```

Comment:
Volat metodu přímo na třídě, to se jen tak nevidí. Víte co to dělá? :)

Reply:
Já jsem nad tímto už přemýšlela a úplně se mi to nelíbí. Teď to vidím tak, že metoda `kill_robot` by měla být metodou robota a ne políčka.